### PR TITLE
Main refactor to decouple core code from pipeline code

### PR DESCRIPTION
**Pull request by victoralfaro-dotcms**
Includes several changes mainly related to reallocate identified CI helper code, mostly written as bash scripts, to its own repository defined as https://github.com/dotCMS/dot-cicd. This why reviewers will find a bunch of missing bash files.

This decoupling give us the benefit of not maintaining pipeline code in an app repository plus the fact that one can change core code and restart individual stages of the travis CI life cycle rather tan restart the whole thing. That's got save us some time (and money).

The reallocated code consist of what it used to be in `core/docker` and `travis` folders.
The integration tests were fixed BTW.

### CODE REVIEW

**Review by dsilvam — APPROVED** [review]

**Review by freddyucv — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java [hunk 1]
```diff
@@ -61,7 +61,7 @@ public DataSource apply() {
                 throw new FileNotFoundException("DB properties file not found");
             }
             return new HikariDataSource(getHikariConfig());
-        } catch (IOException e) {
+        } catch (Exception e) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java:64]: Codacy Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5603107)

File: dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java [hunk 2]
```diff
@@ -78,8 +78,14 @@ File getPropertiesFile() {
 
     @VisibleForTesting
     HikariConfig getHikariConfig() {
+        Logger.info(getClass(), "OJO:>> Instancing HikariConfig with: " + propertiesFile.getPath());
         final HikariConfig config = new HikariConfig(propertiesFile.getPath());
         config.setPoolName(Constants.DATABASE_DEFAULT_DATASOURCE);
+        final StringBuilder sb = new StringBuilder("OJO:>> DB Properties:");
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java:84]: Codacy Issue found: [Avoid variables with short names like sb](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5603107)

Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java:84]: Codacy Issue found: [Avoid variables with short names like sb](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5603107)

File: dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java [hunk 3]
```diff
@@ -8,8 +8,11 @@
 import com.dotmarketing.util.Logger;
 import java.io.File;
 import java.io.FileNotFoundException;
+import java.io.FileReader;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java:11]: Codacy Issue found: [Avoid unused imports such as 'java.io.FileReader'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5603107)

File: dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java [hunk 4]
```diff
@@ -8,8 +8,11 @@
...
 import java.io.IOException;
 import java.net.URL;
+import java.nio.file.Files;
+import java.util.stream.Collectors;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java:15]: Codacy Issue found: [Avoid unused imports such as 'java.util.stream.Collectors'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5603107)

File: dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java [hunk 5]
```diff
@@ -78,8 +81,19 @@ File getPropertiesFile() {
...
     HikariConfig getHikariConfig() {
+        Logger.info(getClass(), "OJO:>> Instancing HikariConfig with: " + propertiesFile.getPath());
+        try {
+            Logger.info(getClass(), String.join("\n", Files.readAllLines(propertiesFile.toPath())));
+        } catch (IOException e) {
+            Logger.error(getClass(), "OJO:>> Error: " + e.getMessage(), e);
+        }
         final HikariConfig config = new HikariConfig(propertiesFile.getPath());
         config.setPoolName(Constants.DATABASE_DEFAULT_DATASOURCE);
+        final StringBuilder sb = new StringBuilder("OJO:>> DB Properties:");
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java:92]: Codacy Issue found: [Avoid variables with short names like sb](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5603107)

File: dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java:11]: Codacy Issue found: [Avoid unused imports such as 'java.io.FileReader'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5603107)

File: dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java:92]: Codacy Issue found: [Avoid variables with short names like sb](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5603107)

File: dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java:11]: Codacy Issue found: [Avoid unused imports such as 'java.io.FileReader'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5603107)

File: dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java:15]: Codacy Issue found: [Avoid unused imports such as 'java.util.stream.Collectors'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5603107)

File: dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java:92]: Codacy Issue found: [Avoid variables with short names like sb](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5603107)

File: dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java:15]: Codacy Issue found: [Avoid unused imports such as 'java.util.stream.Collectors'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5603107)

File: dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java:92]: Codacy Issue found: [Avoid variables with short names like sb](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5603107)

File: dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java:11]: Codacy Issue found: [Avoid unused imports such as 'java.io.FileReader'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5603107)

File: dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java:15]: Codacy Issue found: [Avoid unused imports such as 'java.util.stream.Collectors'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5603107)

File: dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java:92]: Codacy Issue found: [Avoid variables with short names like sb](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5603107)

File: dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java:11]: Codacy Issue found: [Avoid unused imports such as 'java.io.FileReader'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5603107)

File: dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java:15]: Codacy Issue found: [Avoid unused imports such as 'java.util.stream.Collectors'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5603107)

File: dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDataSourceStrategy.java:92]: Codacy Issue found: [Avoid variables with short names like sb](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5603107)

**Review by dsilvam — APPROVED** [review]

**Review by freddyucv — APPROVED** [review]

**Review by nollymar — APPROVED** [review]

**Pull request merged**